### PR TITLE
New methods of building/installing PyThemis

### PR DESCRIPTION
**Pull request by iamnotacake**
Refactor Makefile:
* new target `pythemis_make_wheel` to create a `.whl` Python package, current modern format to be installed into virtual environmants
* new target `pythemis_install_wheel` to install it in currently active virtualenv
* new target `deb_python` that builds a `.deb` package for system-wide installation of PyThemis
* new target `pythemis_install_deb`, alias for `pythemis_deb` + `apt install` of the created pkg
* new target `rpm_python`, similar to `deb_python`
* new target `pythemis_install_rpm`, `pythemis_install_deb`

See cossacklabs/product-docs/pull/317 for related docs update

## Checklist

- [x] Change is covered by automated tests
- [x] ~Benchmark results are attached (if applicable)~
- [x] The [coding guidelines] are followed
- [x] ~Public API has proper documentation~
- [x] ~Example projects and code samples are up-to-date (in case of API changes)~
- [x] Changelog is updated (in case of notable or breaking changes)

[coding guidelines]: https://github.com/cossacklabs/themis/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: .github/workflows/test-python.yaml
Comment (Lagovas) [inline .github/workflows/test-python.yaml:72]: can we install somehow without hardcoding the version? because then we need to update this command for every new release

File: Makefile
Comment (vixentael) [inline Makefile:623]: 👀

File: .github/workflows/test-python.yaml
Comment (iamnotacake) [reply .github/workflows/test-python.yaml]: Done, `make pythemis_install_wheel`

File: Makefile
Comment (Lagovas) [inline Makefile:603]: with this change we will break even our old CI/CD scripts. If we want to remove it and migrate to another approach, we should leave previous targets, notify user's who will use it with next themis version that it's deprecate and address them to the new approach. and after one more release we are free to remove it. And we will have time to update our scripts.

Comment (iamnotacake) [reply Makefile:603]: OK, gonna name it like before. BTW, what do you think about names of all the new `make` targets?

File: Makefile
Comment (Lagovas) [inline Makefile:612]: looks like it's old comment because I don't see anymore patching PATH

File: Makefile
Comment (Lagovas) [inline Makefile:852]: we need add here `libthemis` dependency too

File: Makefile
Comment (Lagovas) [inline Makefile:836]: we already have targets `deb` and `deb_php`. Let's keep naming conventions and name it `deb_python`

File: Makefile
Comment (iamnotacake) [reply Makefile:852]: This also forces user to have Themis itself installed in system as `.deb` package. And it's a bit of an issue imo, because one needs to either add needed repos and install from them or use `make deb` which currently not only builds Themis itself, but also C++ and JNI wrappers. And while I see no problems with the first one, 'cause everybody got a C++ compiler, missing `jni.h` will cause errors and it will fail. So I think it's worth separating wrappers to separate targets like `deb_themispp` and `deb_jni`. To retain backward compatibility we could also move Themis packages to separate targets and make `deb` depend on all of them to work like before.

File: Makefile
Comment (iamnotacake) [reply Makefile]: >why we need python3-six?

Because it's mentioned in `src/wrappers/themis/python/setup.py`. Although, not sure we still need it as it's some [compatibility thing](https://pypi.org/project/six/) to make code with with Python 2 and 3.

File: Makefile
Comment (Lagovas) [reply Makefile:852]: > This also forces user to have Themis itself installed in system as .deb package.

yes, it's okay and expected behavior.

> And it's a bit of an issue imo, because one needs to either add needed repos and install from them or use make deb which currently not only builds Themis itself, but also C++ and JNI wrappers. 

it's okay that it need to install dependencies. If someone wants to install pythemis from the sources, it need to install libthemis too. And as you mentioned, user can do it from the package repository or build with makefile only if he need libthemis from the sources. Yes, for now it requires to compile themispp and themis jni due to `deb` is general target that builds all at once and we don't have `deb_libthemis` or `deb_core` that will not build anything else except core lib. When we will add this target, he will be able to pack deb packet for pythemis and be able to pack .deb libthemis from the sources. Adding one more target to build libthemis deb packet is out of scope this PR and I think we can do it in the separate. But we must specify dependency in the deb metadata for pythemis.

File: Makefile
Comment (Lagovas) [inline Makefile:612]: last comment. can we clean generated artifacts with `clean` target? this command creates new folder with source codes and wheel. or if we can specify destination folder, let's use `$(BIN_PATH)`. It's already mentioned in the `clean` target

**Comment by Lagovas** [comment]
lgtm, it's ready to merge. but let's make all tests green and after that merge.
ping me one more time please to approve when tests will be green

**Comment by Lagovas** [comment]
looks like we need update rust dependencies again. let's up the version to 1.60. looks like a lot of version upgraded their dependencies

**Review by Lagovas — APPROVED** [review]
lgtm, waited green tests)

**Review by G1gg1L3s — APPROVED** [review]
Looks good!

**Review by vixentael — APPROVED** [review]

**Pull request merged**
